spring-cloud-circuitbreaker-hystrix/src/main/java/org/springframework/cloud/circuitbreaker/hystrix/ReactorHystrixCircuitBreaker.java

```diff
@@ -52,6 +54,10 @@ public ReactiveHystrixCircuitBreaker(HystrixObservableCommand.Setter setter) {
 
 	@Override
 	public <T> Flux<T> run(Flux<T> toRun, Function<Throwable, Flux<T>> fallback) {
+		return runFlux(toRun, fallback);
+	}
+
+	private <T> Flux<T> runFlux(Flux<T> toRun, Function<Throwable, Flux<T>> fallback) {
 		HystrixObservableCommand<T> command = createCommand(toRun, fallback);
 
 		return Flux.create(s -> {
```

Comment:
@bsideup @simonbasle how would something like this work with `CorePublisher`?  Without having a concrete type to work with, how would we map things?

Reply:
ping @bsideup @simonbasle ^^^

Reply:
I think something like this could work:

```java
public <T, P extends CorePublisher<T>> P run(P toRun, Function<Throwable, P> fallback)
```

Reply:
@simonbasle but still we have this code `Mono.create` or `Flux.create`, if we are being passed a `CorePublisher` how would we know which one to create, without doing some checks on the types....

Reply:
at least you have a single API and can choose Mono when relevant, via instanceof? all other cases would be turned into Flux? (check the behavior with ParallelFlux though, IIRC it is a CorePublisher too but not a Flux). not 100% ideal I guess